### PR TITLE
[Pal] Fix possible memory leak after calling `handle_deserialize()`

### PR DESCRIPTION
**Pull request by yao-ji**
Signed-off-by: Yao Ji <yao.ji@intel.com>

## How to test this PR? 

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)

### CODE REVIEW

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @yao-ji)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/811#-N8SNTU78Uu97UyoNJqn):*
Remove `possible`. We can do it during final rebase.

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by dimakuv — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @boryspoplawski and @yao-ji)

___
*[`pal/src/host/linux/pal_streams.c` line 284 at r1](https://reviewable.io/reviews/gramineproject/gramine/811#-N8TeeMHEkjcKb7SpUom:-N8Tjpqu-3U2W7HO9CQR:b9qrbhv) ([raw file](https://github.com/gramineproject/gramine/blob/147bc39022a0d6375ae61187ae96b3b49059ee08/pal/src/host/linux/pal_streams.c#L284)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

Is this correct? Isn't `handle` only partially initialized here? This code is quite a mess and hard to follow.
How about instead of adding a close here, we move this header check to before `handle_deserialize`?
</blockquote></details>

Ah, yes, good idea, + 1 to this request.

**Comment by yao-ji** [comment]
*[`pal/src/host/linux/pal_streams.c` line 284 at r1](https://reviewable.io/reviews/gramineproject/gramine/811#-N8TeeMHEkjcKb7SpUom:-N8WSgDv4kuwoj_u4bQU:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/147bc39022a0d6375ae61187ae96b3b49059ee08/pal/src/host/linux/pal_streams.c#L284)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Ah, yes, good idea, + 1 to this request.
</blockquote></details>

Done.

**Review by dimakuv — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @boryspoplawski and @yao-ji)

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by boryspoplawski — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL), "fixup! " found in commit messages' one-liners (waiting on @yao-ji)

**Review by dimakuv — DISMISSED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), "fixup! " found in commit messages' one-liners (waiting on @yao-ji)

**Comment by yao-ji** [comment]
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/811#-N8SNTU78Uu97UyoNJqn):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Remove `possible`. We can do it during final rebase.
</blockquote></details>

Do I need to do the final rebase?

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @boryspoplawski and @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/811#-N93poDc69sXrdSHpQbr:-N93poDc69sXrdSHpQbs:b-pp3vlq) (no related file):*
@yao-ji: In the future, please create PRs from branches in `<your-nick>/<change-descr>` format, even if it's from your private repo - this makes it a bit easier for us to pull and rebase it before merging without polluting our local branch namespaces.

___
___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/811#-N8SNTU78Uu97UyoNJqn):*
<details><summary><i>Previously, yao-ji wrote…</i></summary><blockquote>

Do I need to do the final rebase?
</blockquote></details>

No, it's usually us doing it.

**Comment by mkow** [comment]
Jenkins, retest this please

**Review by boryspoplawski — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: all files reviewed, 1 unresolved discussion, not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/811)* status: :shipit: complete! all files reviewed, all discussions resolved
